src/test/java/task40/SeleniumEasyTests.java

```diff
@@ -0,0 +1,204 @@
+            element.click();
+
+            Assert.assertTrue(element.isSelected());
+        }
+    }
+
+    @Test
+    public void Point6JavaScriptConfirmBoxOkTest() {
+        By clickMeButtonLocator = By.xpath("//button[@onclick = 'myConfirmFunction()']");
+        By statusMessageLocator = By.id("confirm-demo");
+
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        WebElement clickMeButton = driver.findElement(clickMeButtonLocator);
+        clickMeButton.click();
+        Alert alert = driver.switchTo().alert();
+        alert.accept();
+
+        WebElement statusMessage = driver.findElement(statusMessageLocator);
+
+        Assert.assertEquals(statusMessage.getText(), "You pressed OK!");
+    }
+
+    @Test
+    public void Point6JavaScriptConfirmBoxCancelTest() {
+        By clickMeButtonLocator = By.xpath("//button[@onclick = 'myConfirmFunction()']");
+        By statusMessageLocator = By.id("confirm-demo");
+
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        WebElement clickMeButton = driver.findElement(clickMeButtonLocator);
+        clickMeButton.click();
+        Alert alert = driver.switchTo().alert();
+        alert.dismiss();
+
+        WebElement statusMessage = driver.findElement(statusMessageLocator);
+
+        Assert.assertEquals(statusMessage.getText(), "You pressed Cancel!");
+    }
+
+    @Test
+    public void Point6JavaScriptPromptBoxTest() {
+        By clickMeButtonLocator = By.xpath("//button[@onclick = 'myPromptFunction()']");
+        By statusMessageLocator = By.id("prompt-demo");
+        String testMessage = "Test message";
+
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        WebElement clickMeButton = driver.findElement(clickMeButtonLocator);
+        clickMeButton.click();
+        Alert alert = driver.switchTo().alert();
+        alert.sendKeys(testMessage);
+        alert.accept();
+
+        WebElement statusMessage = driver.findElement(statusMessageLocator);
+
+        Assert.assertTrue(statusMessage.getText().contains(testMessage));
+    }
+
+    @Test
+    public void Point7WaitForUserTest() {
+        By getNewUserButtonLocator = By.id("save");
+        By photoLocator = By.cssSelector("#loading img");
+
+        driver.get("https://demo.seleniumeasy.com/dynamic-data-loading-demo.html");
+
+        WebElement getNewUserButton = driver.findElement(getNewUserButtonLocator);
+        getNewUserButton.click();
+
+        wait.until(ExpectedConditions.presenceOfElementLocated(photoLocator));
+    }
+
+    @Test
+    public void Point8RefreshDownloadTest() {
+        driver.get("https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html");
+        WebElement downloadButton = driver.findElement(By.id("cricle-btn"));
+        WebElement downloadBar = driver.findElement(By.cssSelector(".percenttext"));
+
+        downloadButton.click();
+
+        while (true) {
```

Comment:
usually I suggest to add some additional condition to exit from such cycle, because this cycle can theoretically loop and be infinite